langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java

```diff
@@ -213,10 +235,33 @@ private List<String> addAll(List<String> ids, List<Embedding> embeddings, List<T
         return resIds;
     }
 
-    private WeaviateObject buildObject(String id, Embedding embedding, String text) {
+    private WeaviateObject buildObject(String id, Embedding embedding, TextSegment segment) {
         Map<String, Object> props = new HashMap<>();
-        props.put(METADATA_TEXT_SEGMENT, text);
-
+        Map<String, Object> metadata = new HashMap<>();
+        if (segment != null) {
+            props.put(METADATA_TEXT_SEGMENT, segment.text());
+            if (segment.metadata() != null && !segment.metadata().asMap().isEmpty()) {
+                for (Property property : properties) {
+                    if (segment.metadata().containsKey(property.getName())) {
+                        metadata.put(property.getName(), segment.metadata().get(property.getName()));
+                    } else {
+                        metadata.put(property.getName(), "");
+                    }
+                }
+            } else {
+                for (Property property : properties) {
+                    metadata.put(property.getName(), "");
+                }
+                props.put(METADATA, metadata);
+            }
+            props.put(METADATA, metadata);
+        } else {
+            props.put(METADATA_TEXT_SEGMENT, "");
+            for (Property property : properties) {
+                metadata.put(property.getName(), "");
```

Comment:
this logic seems strange, is it actually required?

Reply:
No I should be using "null" instead